spock-core/src/main/java/org/spockframework/lang/SpecInternals.java

```diff
@@ -64,26 +64,25 @@ <T> T oldImpl(T expression) {
   }
 
   <T extends Throwable> T thrownImpl(String inferredName, Class<T> inferredType) {
-    if (inferredType == null) {
-      throw new InvalidSpecException("Thrown exception type cannot be inferred automatically. " +
-        "Please specify a type explicitly (e.g. 'thrown(MyException)').");
-    }
-
     return inferredType.cast(checkExceptionThrown(inferredType));
   }
 
   Throwable thrownImpl(String inferredName, Class<? extends Throwable> inferredType, Class<? extends Throwable> specifiedType) {
-    return checkExceptionThrown(specifiedType);
+    return checkExceptionThrown(specifiedType != null ? specifiedType : inferredType);
```

Comment:
```suggestion
    return checkExceptionThrown(specifiedType == null ? inferredType : specifiedType);
```

Reply:
Just wanted to be consistent with `createMockImpl`, but there it is of course a bit different as there are three alternatives.